src/boost_histogram/_internal/hist.py

```diff
@@ -162,12 +159,24 @@ def _generate_axes_(self):
 
         return AxesTuple(self._axis(i) for i in range(self.ndim))
 
-    def _new_hist(self, _hist):
+    def _new_hist(self, _hist, memo=NOTHING):
```

Comment:
Not sure what this does...

Reply:
If you pass memo, it's a deep copy. (copy and deep copy were separate implementations before). I _think_ memo can be None if this is the outer level of deep copy, so I use a tag here that can't ever be passed as memo.